kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/InternalUtils.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.util;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Optional;
+
+import io.delta.kernel.data.FileDataReadResult;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.fs.FileStatus;
+import io.delta.kernel.utils.CloseableIterator;
+
+import io.delta.kernel.internal.actions.AddFile;
+import io.delta.kernel.internal.data.AddFileColumnarBatch;
+
+public class InternalUtils
```

Comment:
nit: RowUtils?

Reply:
Its more than the row utils. In the future if this ends up too big, we can split it.